cmake/libigl-config.cmake.in

```diff
@@ -2,30 +2,21 @@
 
 include(${CMAKE_CURRENT_LIST_DIR}/libigl-export.cmake)
 
-# Check if Eigen3 target is avaiable, if not try to locate it,
-# either with find_package or with PkgConfig. On success sets
-# LIBIGL_Eigen3_Target to the located Eigen3 target. Of failure
-# the variable is empty.
+# Check if Eigen3 target is avaiable, if not try to locate it
+# with find_package. On success sets LIBIGL_Eigen3_Target to
+# the located Eigen3 target. On failure the variable is empty.
 function(LIBIGL_find_Eigen3)
   set(LIBIGL_Eigen3_Target "" PARENT_SCOPE)
 
   if (TARGET Eigen3::Eigen)
     set(LIBIGL_Eigen3_Target Eigen3::Eigen PARENT_SCOPE)
   else()
-    # First try if Eigen3 can be found with find_package
-    find_package(Eigen3 QUIET)
+    # Try if Eigen3 can be found with find_package
+    find_package(Eigen3 CONFIG QUIET)
     if (NOT Eigen3_FOUND)
-      # find_package failed, next try with PkgConfig
-      find_package(PkgConfig REQUIRED)
-      pkg_check_modules(Eigen3 QUIET IMPORTED_TARGET eigen3)
-
-      if (NOT Eigen3_FOUND)
-        # Both, find_package and PkgConfig failed to locate Eigen3
-        message(FATAL_ERROR "Could not find required dependency Eigen3")
-      else()
-        # PkgConfig succeeded, link against PkgConfig::Eigen3
-        set(LIBIGL_Eigen3_Target PkgConfig::Eigen3 PARENT_SCOPE)
-      endif()
+      message(SEND_ERROR "libigl depends on Eigen3 but Eigen3 could not be found.")
+      # run find_package again to print its error message
+      find_package(Eigen3 CONFIG REQUIRED)
```

Comment:
Isn't that redundant with the line above then?

Reply:
the idea with the extra call to `find_package()` was that the error generated by `find_package` contains some useful information which variables to set in order to help it find Eigen.
The `message(SEND_ERROR "...")` is just there to inform the user why `find_package` is run at all.

Reply:
Why not simply write a `message(STATUS "libigl looking for Eigen3")` or something like that before the first `find_package()`, and make it `REQUIRED`?